service/history/replication/task_processor.go

```diff
@@ -285,12 +298,15 @@ func (p *taskProcessorImpl) processResponse(response *r.ReplicationMessages) {
 
 	scope := p.metricsClient.Scope(metrics.ReplicationTaskFetcherScope, metrics.TargetClusterTag(p.sourceCluster))
 	batchRequestStartTime := time.Now()
+	ctx := context.Background()
 	for _, replicationTask := range response.ReplicationTasks {
+		p.rateLimiter.Wait(ctx)
 		err := p.processSingleTask(replicationTask)
 		if err != nil {
 			// Processor is shutdown. Exit without updating the checkpoint.
 			return
 		}
+		time.Sleep(p.config.ReplicationTaskProcessorTaskProcessingWait(p.shard.GetShardID()))
```

Comment:
Synced offline. Let's change this to a shard level rate limiter.

Reply:
Updated